src/build_workflow/build_target.py

```diff
@@ -41,7 +43,22 @@ def __init__(
     def opensearch_version(self):
         return self.version + "-SNAPSHOT" if self.snapshot else self.version
```

Comment:
Nearly all properties in this class have an `if` statement, code smells like we should create a subclass

Reply:
https://github.com/opensearch-project/opensearch-build/issues/856